### PR TITLE
Add Configurability to Client WebSockets

### PR DESCRIPTION
**Pull request by halilozercan**
**Subsystem**
Client

**Motivation**
Currently there is a small problem with WebSockets feature in ktor-client that originates from non-configurability. Here is a short explanation of how I encountered this issue:

I've been using ktor in both server and client nodes in a side project. I discovered that my client agents are disconnecting due to an internet connectivity issue and not realizing they’ve lost the connection, meanwhile the server is well aware of disconnection.

I’ve been using DefaultWebSocket which means ping-pong should have caught this issue. After some digging, it turns out that ping-pong mechanism actually is not enabled by default on client sockets. The reason is DefaultWebSocket expects a positive value for pingInterval while `WebSockets` is initialized with non-configurable `-1L`. 

Although client WebSockets feature takes pingInterval and maxFrameSize as constructor parameters, install(WebSockets) does not have a configuration option. So, I’m now stuck, not able to turn on a feature that already exists.

Furthermore, maxFrameSize is passed to DefaultWebSocketSession  as
```return DefaultWebSocketSession(this, pingInterval, maxFrameSize)```
it turns out, third parameter is actually timeoutMillis not related to frameSize at all. 

**Solution**
Simply changing the name of maxFrameSize to timeoutMillis and providing a configuration lambda in `prepare` call of the feature gives support for users to enable ping-pong check in their clients. Details of implementation can be found in the commit message.

### CODE REVIEW

**Comment by halilozercan** [comment]
Related documentation PR 

https://github.com/ktorio/ktorio.github.io/pull/263

File: ktor-client/ktor-client-core/common/src/io/ktor/client/features/websocket/WebSockets.kt [hunk 1]
```diff
@@ -15,20 +15,22 @@ import io.ktor.util.*
...
- * @property maxFrameSize - max size of single websocket frame.
+ * @property pingIntervalMillis - interval between [FrameType.PING] messages.
+ * @property timeoutMillis - timeout to receive a [FrameType.PONG] message.
  */
 @KtorExperimentalAPI
 @OptIn(WebSocketInternalAPI::class)
 class WebSockets(
-    val pingInterval: Long = -1L,
-    val maxFrameSize: Long = Int.MAX_VALUE.toLong()
+    var pingIntervalMillis: Long = -1L,
```
Comment (e5l) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/features/websocket/WebSockets.kt]: Consider keeping old names for now, we can't change them until ktor `1.4.0`

Comment (halilozercan) [reply ktor-client/ktor-client-core/common/src/io/ktor/client/features/websocket/WebSockets.kt]: As I understand, we should keep the old names and default values but `maxFrameSize` would still be passed as timeoutMillis?

**Comment by halilozercan** [comment]
I think changing variable names back to what they were would defeat the purpose of the fix. I'm going to reopen this pull request with `1.4.0` as destination.

**Pull request closed without merging**